src/authentication.js

```diff
@@ -16,11 +19,19 @@ module.exports = (app) => {
   // to create a new valid JWT (e.g. local or oauth2)
   app.service('authentication').hooks({
     before: {
-      // Having issues with authenticating? Try changing the order of these two hooks
-      create: [isVerified(), authentication.hooks.authenticate(config.strategies)],
+      create: [
+        validateSchema(loginSchema),
+        authentication.hooks.authenticate(config.strategies),
+        isVerified(),
+      ],
       remove: [
         authentication.hooks.authenticate('jwt'),
       ],
     },
+    error(context) {
+      if (context.error.name === 'Error') {
+        throw new NotAuthenticated('User\'s email is not yet verified.');
```

Comment:
Are you sure this will always be caused by an unverified email and not other errors? Basically this could swallow other errors making debugger hard